src/mgmt/srv_drpc.c

```diff
@@ -66,6 +66,43 @@ pack_daos_response(Mgmt__DaosResp *daos_resp, Drpc__Response *drpc_resp)
 	drpc_resp->body.data = body;
 }
 
+void
+ds_mgmt_drpc_prep_shutdown(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PrepShutdownReq	 *req = NULL;
+	Mgmt__DaosResp		 *resp = NULL;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__prep_shutdown_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILURE;
+		D_ERROR("Failed to unpack req (prep shutdown)\n");
+		return;
+	}
+
+	D_INFO("Received request to prep shutdown %u\n", req->rank);
+
+	D_ALLOC_PTR(resp);
```

Comment:
Suggestion - could make the Mgmt__DaosResp statically allocated instead. I've been taking that approach for many of the response structs, gives us fewer things to clean up.

Reply:
https://github.com/daos-stack/daos/pull/1591